### ISSUE TITLE
Create LLM-based reranker

### ISSUE THREAD

**Issue by AsphodelRem** [AsphodelRem/RAGU#55]
Add an LLM reranker that scores (query, document) pairs using next-token logits.

Example:

#### Prompt template:
"""
Query: {query}
Document: {document}

Relevant:
"""

Compute relevance score from logits:
score = log P(" yes")

or 

score = log P("yes") - log P("no")
